QHyper/solvers/QAOA/Pennylane.py

```diff
@@ -0,0 +1,236 @@
+import pennylane as qml
+import numpy as np
+
+from .parser import parse_hamiltonian
+from ..solver import Solver
+from ...problems.problem import Problem
+from ...optimizers.optimizer import Optimizer, HyperparametersOptimizer
+
+
+class PennyLaneQAOA(Solver):
+    """QAOA solver based on PennyLane
+
+    Attributes
+    ----------
+    problem : Problem
+        object of class Problem, defining problem
+    angles: list[float]
+        angles for QAOA, also initial values for optimizers
+    optimizer : Optmizer
+        optimizer that will be used to optimize angles (default None)
+    layers : int
+        number of layers in circuit (default 3)
+    mixer : str
+        mixer name (currently only "X" is supported) (default "X")
+    weights : list[float]
+        needed for converting Problem to QUBO, if QUBU is already provided
+        weights should be [1] (default None)
+    hyperoptimizer : HyperparametersOptimizer
+        optimizer to tune the weights (default None)
+    backend : str
+        name of the backend (default "default.qubit")
+    dev : qml.device
+        PennyLane quantum device
+    """
+
+    def __init__(self, **kwargs) -> None:
+        """
+        Parameters
+        ----------
+        problem : Problem
+            object of class Problem, defining problem
+        angles: list[float]
+            angles for QAOA, also initial values for optimizers
+        optimizer : Optmizer
+            optimizer that will be used to optimize angles (default None)
+        layers : int
+            number of layers in circuit (default 3)
+        mixer : str
+            mixer name (currently only "X" is supported) (default "X")
+        weights : list[float]
+            needed for converting Problem to QUBO, if QUBU is already provided
+            weights should be [1] (default None)
+        hyperoptimizer : HyperparametersOptimizer
+            optimizer to tune the weights (default None)
+        backend : str
+            name of the backend (default "default.qubit")
+        """
+
+        self.problem: Problem = kwargs.get("problem")
+        self.angles: list[float] = kwargs.get("angles")
+
+        self.optimizer: Optimizer = kwargs.get("optimizer", None)
+        self.layers: int = kwargs.get("layers", 3)
+        self.mixer: str = kwargs.get("mixer", "X")
+        self.weights: list[float] = kwargs.get("weights", None)
+        self.hyperoptimizer: HyperparametersOptimizer = kwargs.get("hyperoptimizer", None)
+        self.backend: str = kwargs.get("backend", "default.qubit")
+
+        self.dev = qml.device(self.backend, wires=self.problem.wires)
+
+    def _create_cost_operator(self, weights) -> qml.Hamiltonian:
+        ingredients = [self.problem.objective_function] + self.problem.constraints
```

Comment:
Maybe rename ingredient to element (as an element in math)?

Reply:
I though the `ingredients` are in sum, what about `term` instead? All of these elements are part of the sum